src/matrixroomhandler.ts

```diff
@@ -182,8 +182,8 @@ export class MatrixRoomHandler {
                     regexp: "[A-Za-z0-9_\-]{2,100}",
                 },
                 discriminator: {
-                    placeholder: "1234",
-                    regexp: "[0-9]{4}",
+                    placeholder: "0",
+                    regexp: "[0-9]{1,4}",
```

Comment:
This regexp allows 2- and 3-digit values, when AIUI the discriminator can only be either a 4-digit value or 0.  If that's the case, a regexp of `0|[0-9]{4}` should do the trick.

Reply:
It should be treated as a string. According to reverse engineering work done for https://github.com/spacebarchar/server, Discord's client also treats it as a string; and Discord itself likely also stores it as a string. I have previously tested alphanumeric discriminators against the Discord client and it worked fine. Discord discriminators should really be treated as opaque strings, just like Matrix room IDs.

Reply:
Discord and Matrix implementations currently send fixed length values for normally created Matrix rooms (not with a manually overridden room ID) and normally created Discord users (not with a manually overridden discriminator), but this is not guaranteed in the API.

Reply:
@AndrewFerr already signed-off, see the commit message